filament/backend/src/CommandStream.cpp

```diff
@@ -75,9 +75,12 @@ void CommandStream::execute(void* buffer) {
 
     Driver& UTILS_RESTRICT driver = *mDriver;
     CommandBase* UTILS_RESTRICT base = static_cast<CommandBase*>(buffer);
-    while (UTILS_LIKELY(base)) {
-        base = base->execute(driver);
-    }
+
+    driver.execute([&base, &driver] () {
+        while (UTILS_LIKELY(base)) {
+            base = base->execute(driver);
+        }
+    });
```

Comment:
I *believe* here you can make the lambda `mutable` and take `base` by value (because the updated `base` is not needed in this function once we return). This will be more efficient. If this doesn't work, it would be better to capture `buffer`.  Same thing for `driver`, better to capture `mDriver`.

Basically, we don't want to capture things by reference, because this will be slower (i.e. an extra dereference).

Reply:
I can't capture `mDriver` directly, but I can do something like this:

```
    mDriver->execute([base, driver = this->mDriver] () mutable {
        while (UTILS_LIKELY(base)) {
            base = base->execute(*driver);
        }
    });
```

Reply:
Actually thinking more about it, you don't want to use captured variables in the loop -- this is because they are actually a class member, and often the compiler can't keep them in registers. Just to be safe, I think we should do this:

```
driver.execute([this, buffer]() {
    Driver& UTILS_RESTRICT driver = *mDriver;
    CommandBase* UTILS_RESTRICT base = static_cast<CommandBase*>(buffer);
    while (UTILS_LIKELY(base)) {
        base = base->execute(driver);
    }
});
```